milabench/utils.py

```diff
@@ -102,3 +108,92 @@ def make_constraints_file(pth, constraints):
         return (pth,)
     else:
         return ()
+
+
+def discover_validation_layers(module):
+    """Discover validation layer inside the milabench.validation module"""
+    path = module.__path__
+    name = module.__name__
+
+    layers = {}
+
+    for _, layerpath, _ in pkgutil.iter_modules(path, name + "."):
+        layername = layerpath.split(".")[-1]
+        layermodule = importlib.import_module(layerpath)
+
+        if hasattr(layermodule, "_Layer"):
+            layers[layername] = layermodule._Layer
+
+    return layers
+
+
+VALIDATION_LAYERS = discover_validation_layers(milabench.validation)
+
+
+def available_layers():
+    return VALIDATION_LAYERS.keys()
+
+
+def validation_layers(*layer_names, **kwargs):
+    """Initialize a list of validation layers"""
+    layers = []
+
+    for layer_name in layer_names:
+        layer = VALIDATION_LAYERS.get(layer_name)
+
+        if layer is not None:
+            layers.append(layer())
+        else:
+            names = list(VALIDATION_LAYERS.keys())
+            raise RuntimeError(f"Layer `{layer_name}` does not exist: {names}")
+
+    return layers
+
+
+class _LoggerProxy:
+    def __init__(self, funs) -> None:
+        self.funs = funs
+
+    def __call__(self, *args: Any, **kwds: Any) -> Any:
+        for _, fun in self.funs.items():
+            try:
+                fun(*args, **kwds)
+
+            except Exception:
+                logger_name = getattr(fun, "__name__", fun)
+                print(f"Error happened in logger {logger_name}", file=sys.stderr)
+                print("=" * 80)
+                traceback.print_exc()
+                print("=" * 80)
```

Comment:
The =s should be printed to stderr as well (that was also an oversight in master currently, I realize)

Reply:
done